guides/mutations/mutation_errors.md

```diff
@@ -0,0 +1,129 @@
+---
+layout: guide
+search: true
+section: Mutations
+title: Mutation errors
+desc: Tips for handling and returning errors from mutations
+index: 2
+---
+
+How can you handle errors inside mutations? Let's explore a couple of options.
+
+## Raising Errors
+
+One way to handle an error is by raising, for example:
+
+```ruby
+def resolve(id:, attributes:)
+  # Will crash the query if the data is invalid:
+  Post.find(id).update!(attributes.to_h)
+  # ...
+end
+```
+
+Or:
+
+```ruby
+def resolve(id:, attributes:)
+  if post.update(attributes)
+    { post: post }
+  else
+    raise GraphQL::ExecutionError, post.errors.full_messages.join(", ")
+  end
+end
+```
+
+This kind of error handling _does_ express error state (either via `HTTP 500` or by the top-level `"errors"` key), but it doesn't take advantage of GraphQL's type system and can only express one error at a time. It works, but a stronger solution is to treat errors as data.
+
+## Errors as Data
+
+Another way to handle rich error information is to add _error types_ to your schema, for example:
+
+```ruby
+class Types::UserError < Types::BaseObject
+  description "A user-readable error"
+
+  field :message, String, null: false,
+    description: "A description of the error"
+  field :attribute, String, null: false,
```

Comment:
Have you considered advising to use `[String!]` instead of `String!` here to future proof for nested arguments.

We may also want to advise using `path` as a field name instead of `attribute`. The `path` would be scoped to the current field.

We could also include a request/response example like this:

```graphql
mutation {
  createPost(title: "", body: "") {
    post {
      id
    }
    errors {
      message
      path
    }
  }
}
```

```json
{
  "data": { 
    "createPost": {
      "post": null,
      "errors": [
        { "message": "Title can't be blank", "path": ["title"] },
        { "message": "Title can't be blank", "path": ["body"] }
      ]
    }
  }
}
```

Reply:
:+1: sure, I'll update that